### DISCUSSION TITLE
How to create a Platform specific MessageDialogService thats injected to ViewModels, ViewModels that are used for both Avalonia and WPF (or some other Platform)?

### DISCUSSION THREAD

**Discussion by EtherGhost** [AvaloniaUI/Avalonia#6384] [Q&A] [answered]
I'm using the same ViewModels for Avalonia and WPF. How can I create a Platform specific MessageDialogService that's injected to my ViewModels, ViewModels that are used for both Avalonia and WPF?

The question may not be specific to Avalonia but I think there are other cases out there where apps use Avalonia for desktop and eg Uno Platform for mobile devices and where ViewModels are shared.

**Comment by EtherGhost** [comment, marked as answer]
I have created a small Project:
https://github.com/kvalitetskontrollMarkaryd/AvaloniaWpfMessageDialogService.git

**Reply by EtherGhost** [reply]
With help from Gitter I have updated the small project and it's working now. It uses standard WPF MessageBox and MessageBox.Avalonia.
Problem is that Avalonia MessageBox is returning a Task and WPF MessageBox not. 
I will leave it like this for now but if someone would like to enhance it, go for it! 
Thanks to @Fido789 at Gitter Avalonia for the help!

**Reply by maxkatz6** [reply]
Sync dialogs is a problem in cross platform world. So, if you can, prefer having async API for dialogs.

**Reply by EtherGhost** [reply]
The problem is to wait for the Task result when fired from a sync method / event. Eg. OnClosing event or a command execute.
